src/llmq/quorums_signing.h

```diff
@@ -66,9 +79,16 @@ class CRecoveredSig
 // TODO implement caching to speed things up
 class CRecoveredSigsDb
 {
+    static const size_t MAX_CACHE_SIZE = 30000;
+    static const size_t MAX_CACHE_TRUNCATE_THRESHOLD = 50000;
+
 private:
     CDBWrapper db;
 
+    CCriticalSection cs;
+    std::unordered_map<std::pair<Consensus::LLMQType, uint256>, std::pair<bool, int64_t>> hasSigForIdCache;
```

Comment:
`has` part is confusing imo.. why not `map`?

same for the second one

Reply:
`has` comes from the method name for which this map caches results.